### PR TITLE
Fix #10429: Kafka Sample data improvements and adding support for JSONSchema and Protobuf

### LINKED ISSUE #10429: Kafka Sample data improvements and adding support for JSONSchema and Protobuf
**Affected module**
Ingestion

**Describe the bug**
Kafka sample data should look at the latest 10-50 messages and parse if they avro or string 

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by harshach**
### Describe your changes :

I worked on the ..... because ...

#
### Type of change :

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by harshach** [comment]
@ulixius9 @OnkarVO7 , parsing kafka's protobuf schema resulting in the following error

```
 DEBUG    {metadata.Ingestion:protobuf_parser:162} - Traceback (most recent call last):                   15:43  harsha@Sriharshas-MacBook-Pro
  File "/Users/harsha/Code/OpenMetadata/ingestion/src/metadata/parsers/protobuf_parser.py", line 158, in get_protobuf_python_object
    class_ = getattr(message, snake_to_camel(self.config.schema_name))
AttributeError: module 'protobuf_test_pb2' has no attribute 'ProtobufTest'

[2023-03-03 15:41:37] WARNING  {metadata.Ingestion:protobuf_parser:163} - Unable to create protobuf python module for protobuf_test: module 'protobuf_test_pb2' has no attribute 'ProtobufTest'
[2023-03-03 15:41:37] DEBUG    {metadata.Ingestion:protobuf_parser:187} - Traceback (most recent call last):
  File "/Users/harsha/Code/OpenMetadata/ingestion/src/metadata/parsers/protobuf_parser.py", line 179, in parse_protobuf_schema
    field_models = self.get_protobuf_fields(instance.DESCRIPTOR.fields)
AttributeError: 'NoneType' object has no attribute 'DESCRIPTOR'

[2023-03-03 15:41:37] WARNING  {metadata.Ingestion:protobuf_parser:188} - Unable to parse protobuf schema for protobuf_test: 'NoneType' object has no attribute 'DESCRIPTOR'
[2023-03-03 15:41:37] DEBUG    {metadata.Ingestion:metadata_rest:126} - Processing Create request <class 'metadata.generated.schema.api.data.createTopic.CreateTopicRequest'>
```

Any ideas on how we parse the protobuf contents dynamically using the schema? 

@chirag-madlani @Sachin-chaurasiya, we should display both schemaFields and schemaText. When schemaFields returns empty, and schemaText is available, we should display that.
image
image

We should support display protobuf contents as well

**Comment by harshach** [comment]
for a avrò record like this

```
{
  "namespace": "io.codebrews.customer",
  "type": "record",
  "name": "Customer",
  "doc": "Its a customer record.",
  "fields": [
    {
      "name": "firstName",
      "type": "string",
      "doc": "first name of the customer."
    },
    {
      "name": "lastName",
      "type": "string"
    },
    {
      "name": "product",
      "type": "string"
    }
  ]
}

```
we are ignoring the top level "Record" and instead of creating a single field of Customer Record and nested fields
we create a flat structure like this
image

We also need to extract doc values out of these Avro schemas and attach them as descriptions. I added support to that but the parsing needs to create a nested structure of the Avro schema

**Comment by chirag-madlani** [comment]
@harshach UI changes pushed. I have one query Is it possible that we won't be having any of these values? `schemaFields` or `schemaText`?

- If `schemaFields` is empty
Screenshot 2023-03-06 at 8 02 58 PM

- With switch for Fields & Text
Screenshot 2023-03-06 at 8 07 45 PM
Screenshot 2023-03-06 at 8 07 38 PM

**Review by ShaileshParmar11 — APPROVED** [review]
UI side changes looks good

**Comment by harshach** [comment]
@ulixius9 @OnkarVO7 any update on the Avro parsing?

**Comment by OnkarVO7** [comment]
> @ulixius9 @OnkarVO7 any update on the Avro parsing?

@harshach avro parsing is handled from top level and it is updated in the PR. We are good from ingestion side too

**Comment by chirag-madlani** [comment]
@harshach fix case for no schema present 
Screenshot 2023-03-07 at 10 57 10 AM

File: ingestion/src/metadata/ingestion/source/messaging/common_broker_source.py [hunk 1]
```diff
@@ -46,6 +47,15 @@
 logger = ingestion_logger()
 
 
+def on_assign(consumer, partitions):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/messaging/common_broker_source.py:50]: nit - maybe add a more meaningful name? `assign_partitions_to_consumer`?

**Review by pmbrull — APPROVED** [review]
lgtm, one nit comment if you have the time

**Comment by harshach** [comment]
can't figure out the python checkstyle
```
pycln ingestion/ openmetadata-airflow-apis/ --extend-exclude ingestion/src/metadata/generated
Looks good! ✨ 🍰 ✨
857 files left unchanged.
isort ingestion/ openmetadata-airflow-apis/ --skip ingestion/src/metadata/generated --skip ingestion/env --skip ingestion/build --skip openmetadata-airflow-apis/build --profile black --multi-line 3
Skipped 2 files
black ingestion/ openmetadata-airflow-apis/ --extend-exclude ingestion/src/metadata/generated
All done! ✨ 🍰 ✨
857 files left unchanged.
```

**Comment by OnkarVO7** [comment]
@harshach python checkstyle should be fixed now

**Review by pmbrull — APPROVED** [review]

**Pull request merged**